### PR TITLE
adjustable retention and loglevel in logback.xml

### PR DESCRIPTION
**Pull request by teplydat**
| Q               | A
| --------------- | ---
| Bug fix?        | no?
| New feature?    | yes
| API breaks?     | no
| Deprications?   | no
| Related tickets | -
| License         | Apache 2.0

### What's in this PR?
Possibility to change retention and loglevel in logback.xaml via CR.

### Why?
Retention time of logfiles and loglevels are currently hardcoded in https://github.com/Orange-OpenSource/nifikop/blob/master/pkg/resources/templates/config/logback.xml.go

### Additional context
In case of no additional values in CR: default values are used. CR is also not extended (keeps it clean).
Usage to overwrite defaults via CR:
```
apiVersion: nifi.orange.com/v1alpha1
kind: NifiCluster
spec:
  logback:
    maxHistory:
      appFile: 30
      userFile: 30
      bootstrapFile: 5
    maxFileSize: 
      appFile: "100MB"
    logLevel:
      nifi: "INFO"
      nifiProcessors: "WARN"
      nifiProcessorsStandardLogAttribute: "INFO"
      nifiProcessorsStandardLogMessage: "INFO"
      nifiControllerRepositoryStandardProcessSession: "WARN"
      zookeeperClientCnxn: "ERROR"
      zookeeperServerNIOServerCnxn: "ERROR"
      zookeeperServerNIOServerCnxnFactory: "ERROR"
      zookeeperServerQuorum: "ERROR"
      zookeeperZooKeeper: "ERROR"
      zookeeperServerPrepRequestProcessor: "ERROR"
      calciteRuntimeCalciteException: "OFF"
      curatorFrameworkRecipesLeaderLeaderSelector: "OFF"
      curatorConnectionState: "OFF"
      nifiCluster: "DEBUG"
      nifiServerJettyServer: "INFO"
      jetty: "INFO"
      springframework: "ERROR"
      jerseyInternalErrors: "ERROR"
      nifiWebSecurity: "INFO"
      nifiWebApiConfig": "INFO"
      nifiAuthorization: "INFO"
      nifiClusterAuthorization: "INFO"
      nifiWebFilterRequestLogger: "INFO"
      nifiBootstrap: "INFO"
      nifiBootstrapCommand: "INFO"
      nifiStdOut: "INFO"
      nifiStdErr: "ERROR"
      root: "INFO"
```
Specify only configs that should be other than default.

### Checklist

- [x] Implementation tested
- [ ] Comments as documentation added
- [ ] Error handling code meets the [guideline](docs/error-handling-guide.md)
- [ ] Logging code meets the guideline
- [ ] User guide and development docs updated (if needed)
- [ ] Append changelog with changes

### To Do
- [ ] Do you think it's an appropriate solution?

Basically we would prefer something like:
```
apiVersion: nifi.orange.com/v1alpha1
kind: NifiCluster
spec:
  logconfigoverride:
    configMap: logback
    namespace: default
```
but the logback.xml is already in a confimap that contains multiple config files (fully managed by the operator).

### CODE REVIEW

**Comment by teplydat** [comment]
Hi @erdrix 

what do you think about? This is just a draft how to be able to change log configs.

**Comment by erdrix** [comment]
Thaks for the suggestion, I think providing the information using a configmap or a secret is more suitable for logback.xml than using a template (because the content of the file is not really stable). 

If you have a bit of time, you can take a look of the PR #87  where I implemented the logic to pass a configmap or secret data to replace the default configuration !
